File: src/action/ActionHeal.java
package com.leekwars.game.fight.action;

import com.alibaba.fastjson.JSONArray;
import com.leekwars.game.fight.entity.Entity;

public class ActionHeal implements Action {

	private final int target;
	private final int life;

	public ActionHeal(Entity target, int life) {
		this.target = target.getFId();
		this.life = life;
	}

	@Override
	public JSONArray getJSON() {
		JSONArray retour = new JSONArray();
		retour.add(Action.HEAL);
		retour.add(target);
		retour.add(life);
		return retour;
	}
}

File: src/action/ActionHeal.cpp
#include "ActionHeal.hpp"
#include "../entity/Entity.hpp"

ActionHeal::ActionHeal(Entity* target, int life) {
	this->target = target->id;
	this->life = life;
}

Json ActionHeal::json() const {
	return {(int) ActionType::HEAL, target, life};
}
